pkg/processor/instance.go

```diff
@@ -42,37 +43,50 @@ type (
 	ProvisionType int
 )
 
-// Interface is the interface that represents a single message processor that
-// can be executed on one record and manipulate it.
-type Interface interface {
-	// Process runs the processor function on a record.
-	Process(ctx context.Context, record record.Record) (record.Record, error)
-
-	// InspectIn starts an inspection session for input records for this processor.
-	InspectIn(ctx context.Context, id string) *inspector.Session
-	// InspectOut starts an inspection session for output records for this processor.
-	InspectOut(ctx context.Context, id string) *inspector.Session
-
-	// Close closes this processor and releases any resources
-	// which may have been used by it.
-	Close()
+type Registry interface {
+	Get(ctx context.Context, pluginName string, id string) (sdk.Processor, error)
```

Comment:
A bit of a nitpick - I think we should move this to `service.go`, since it's used in that file. Same goes for the `go:generate` statement on top.

Reply:
Btw, maybe we should call this `PluginRegistry`? 🤔